extensions/mssql/src/objectExplorerNodeProvider/connection.ts

```diff
@@ -6,226 +6,93 @@
-			options: {}
-		};
-		return info;
-	}
-
-	public get connectionId(): string {
-		return this._connectionId;
-	}
-
-	public get host(): string {
-		if (!this._host) {
-			this.ensureHostAndPort();
-		}
-		return this._host;
-	}
+	private _port: string;
+	private _user: string;
+	private _pass: string;
 
-	/**
-	 * Sets host and port values, using any ',' or ':' delimited port in the hostname in
-	 * preference to the built in port.
-	 */
-	private ensureHostAndPort(): void {
-		this._host = this.connectionInfo.options[constants.hostPropName];
-		this._knoxPort = Connection.getKnoxPortOrDefault(this.connectionInfo);
-		// determine whether the host has either a ',' or ':' in it
-		this.setHostAndPort(',');
-		this.setHostAndPort(':');
-	}
-
-	// set port and host correctly after we've identified that a delimiter exists in the host name
-	private setHostAndPort(delimeter: string): void {
-		let originalHost = this._host;
-		let index = originalHost.indexOf(delimeter);
-		if (index > -1) {
-			this._host = originalHost.slice(0, index);
-			this._knoxPort = originalHost.slice(index + 1);
-		}
-	}
-
-	public get user(): string {
-		return this.connectionInfo.options[constants.userPropName];
-	}
-
-	public get password(): string {
-		return this.connectionInfo.options[constants.passwordPropName];
-	}
-
-	public get knoxport(): string {
-		if (!this._knoxPort) {
-			this.ensureHostAndPort();
+	constructor(connectionInfo: sqlops.connection.Connection | sqlops.IConnectionProfile) {
+		if (!this.isValid(connectionInfo)) {
+			throw new Error(localize('invalidConnectionInfo', 'Invalid ConnectionInfo is provided.'));
 		}
-		return this._knoxPort;
-	}
-
-	private static getKnoxPortOrDefault(connInfo: sqlops.ConnectionInfo): string {
-		let port = connInfo.options[constants.knoxPortPropName];
-		if (!port) {
-			port = constants.defaultKnoxPort;
-		}
-		return port;
-	}
-
-	public get connecttimeout(): number {
-		let timeoutSeconds: number = this.connectionInfo.options['connecttimeout'];
-		if (!timeoutSeconds) {
-			timeoutSeconds = constants.hadoopConnectionTimeoutSeconds;
+		if ('id' in connectionInfo) {
+			this._sqlClusterConnProfile = connectionInfo;
+			this._sqlClusterConnObj = this.toConnObj(this._sqlClusterConnProfile);
+		} else {
+			this._sqlClusterConnObj = connectionInfo;
+			this._sqlClusterConnProfile = this.toConnProfile(this._sqlClusterConnObj);
 		}
-		// connect timeout is in milliseconds
-		return timeoutSeconds * 1000;
+		this._host = this._sqlClusterConnObj.options[constants.hostPropName];
+		this._port = this._sqlClusterConnObj.options[constants.knoxPortPropName];
+		this._user = this._sqlClusterConnObj.options[constants.userPropName];
+		this._pass = this._sqlClusterConnObj.options[constants.passwordPropName];
 	}
 
-	public get sslverification(): string {
-		return this.connectionInfo.options['sslverification'];
-	}
+	public get sqlClusterConnObj(): sqlops.connection.Connection { return this._sqlClusterConnObj; }
```

Comment:
Please rename to `sqlClusterConnection`, and do the same for private field and related functions.